packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -3072,14 +3084,24 @@ bool debugIsLocalCreationLocation(Object object) {
   bool isLocal = false;
   assert(() {
     final _Location? location = _getCreationLocation(object);
-    if (location == null)
-      isLocal =  false;
-    isLocal = WidgetInspectorService.instance._isLocalCreationLocation(location);
+    if (location != null) {
+      isLocal = WidgetInspectorService.instance._isLocalCreationLocation(location.file);
+    }
     return true;
   }());
   return isLocal;
 }
 
+/// Returns true if a [Widget] is user created.
+///
+/// This is a faster variant of `debugIsLocalCreationLocation` that is available
+/// in debug and profile builds.
```

Comment:
the docs should explain why we have `debugIsLocalCreationLocation` (right now the docs suggest this version is faster and available more places, which seems net better in every way)

Reply:
Can we also clarify whether user created means that the user has defined this widget, or whether it means that user code explicitly instantiated this widget?

i.e.

```
// file: main.dart

class MyWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) => Container();
}
```

Is MyWidget user created? Is the container in there user created?

Reply:
> the docs should explain why we have debugIsLocalCreationLocation (right now the docs suggest this version is faster and available more places, which seems net better in every way)

Done, `debugIsLocalCreationLocation` takes Object as a parameter and could technically be expanded in the future.  `debugIsWidgetLocalCreationLocation` is just for Widget.

Reply:
> Can we also clarify whether user created means that the user has defined this widget, or whether it means that user code explicitly instantiated this widget?

Done, defined in the docstring for the flag.